app/forms/user_phone_form.rb

```diff
@@ -30,31 +30,59 @@ def submit(params)
     FormResponse.new(success: success, errors: errors.messages, extra: extra_analytics_attributes)
   end
 
-  def phone_changed?
-    formatted_user_phone != phone
+  def delivery_preference_sms?
+    return true if phone_configuration.blank?
+    phone_configuration&.delivery_preference == 'sms'
+  end
+
+  def delivery_preference_voice?
+    phone_configuration&.delivery_preference == 'voice'
+  end
+
+  def phone_config_changed?
+    return true if formatted_user_phone != phone
+    return true if phone_configuration&.delivery_preference != otp_delivery_preference
+    return true if otp_make_default_number && !default_phone_configuration?
```

Comment:
What is the reason for making them reconfirm the phone if they want it to be the default?

Reply:
We had a brief discussion about this during our first code review. I believe you said that this is the convention we currently had in place for any changes to the phone configuration. However, it seems like a good idea to re-confirm before the user makes it their default.

Reply:
It would be simple to save without confirmation if only default is chosen. 

Here are the two scenarios.
1. Only default is selected: 
- We save default and return to account page.
2. Default is selected and Phone Number or Delivery Preference is changed
- We confirm the phone number. No change (Including Default) is saved until after confirmation is complete

Reply:
Gotcha, yeah. Not a big deal here, but may be worth a follow-on story?